rpp30-boc-meltdown/biteshare
# Add a reusable button

## Description

I created a reusable button component to keep our buttons on different screen consistent. Also added the mock data for restaurants with menus.
Feel free to change anything!

## How to test

Pull this branch and spin up the simulator, a button should be there.
Clicking the button should switch the role to be either 'host' or 'guest' (as an example of updating state)

## Notes

The default button looks like this: 
![Screen Shot 2021-12-21 at 1 24 12 PM](https://user-images.githubusercontent.com/43324065/147001493-859400a3-3a67-4899-927a-a1563dd89bf9.png)
Passing in a prop called `buttonStyle` could overwrite some of the preset styles for the button:
e.g. `buttonStyle={{ backgroundColor: colors.brand.beachLight }}` changed the button to
![Screen Shot 2021-12-21 at 1 24 28 PM](https://user-images.githubusercontent.com/43324065/147001615-4cfc88be-add1-4de5-82c9-4818173871d3.png)
Passing in a prop called `textStyle` could overwrite some of the preset styles for the text:
e.g. `textStyle={{ color: '#fff' }}` changed the button to
![Screen Shot 2021-12-21 at 1 24 53 PM](https://user-images.githubusercontent.com/43324065/147001707-4da54613-3420-473f-92db-3fafebf31dc6.png)
It also accepts `size` (default to 100px, width and height change dynamically according to this size), `title` (the button text) and `onPress` (the click handler for the button).

## Issue tracking

N/A
